src/platform_impl/linux/util.rs

```diff
@@ -70,3 +74,46 @@ pub fn set_size_constraints<W: GtkWindowExt + WidgetExt>(
     geom_mask,
   )
 }
+
+pub struct WindowMaximizeProcess<W: GtkWindowExt + WidgetExt> {
+  window: W,
+  maximized: bool,
+  resizable: bool,
+  step: u8,
+}
+
+impl<W: GtkWindowExt + WidgetExt> WindowMaximizeProcess<W> {
+  pub fn new(window: W, maximized: bool, resizable: bool) -> Rc<RefCell<Self>> {
+    Rc::new(RefCell::new(Self {
+      window,
+      maximized,
+      resizable,
+      step: 0,
+    }))
+  }
+
+  pub fn next_step(&mut self) -> glib::ControlFlow {
+    match self.step {
+      0 => {
+        if self.maximized {
+          self.window.set_resizable(true);
+          self.step += 1;
+          glib::ControlFlow::Continue
+        } else {
+          self.step = 2;
+          glib::ControlFlow::Continue
```

Comment:
should we just break here? why we go to step 2 when we didn't change anything?

Reply:
That was necessary because I removed `window.set_resizable(attributes.resizable)` when creating the new window.

```rust
    let maximize_process =
      util::WindowMaximizeProcess::new(window.clone(), attributes.maximized, attributes.resizable);
    glib::idle_add_local_full(glib::Priority::HIGH_IDLE, move || {
      let mut maximize_process = maximize_process.borrow_mut();
      maximize_process.next_step()
    });

    //window.set_resizable(attributes.resizable); -> remove this
    window.set_deletable(attributes.closable);
```

If we go with this approach:

```suggestion
      glib::ControlFlow::Break
```

It seems like we’ll need to add `window.set_resizable(attributes.resizable);` back again.

Reply:
yeah let's add that back and tweak the logic a bit to avoid unnecessary work, like if `attributes.maximized` is `false`, just skip the logic

Reply:
actually let's do it like this
```rs
if attributes.maximized {
  // glib::idle_add_local ...etc
} else {
  window.set_resizable(attributes.resizable);
}
```
and keep `glib::ControlFlow::Continue` and just add note why we are doing that instead of breaking